pageserver/src/http/routes.rs

```diff
@@ -1900,6 +1920,9 @@ pub fn make_router(
             "/v1/tenant/:tenant_id/timeline/:timeline_id/layer/:layer_file_name",
             |r| api_handler(r, evict_timeline_layer_handler),
         )
+        .post("/v1/tenant/:tenant_shard_id/heatmap_upload", |r| {
+            api_handler(r, secondary_upload_handler)
+        })
```

Comment:
Originally I had this in the secondary prefix, but it felt awkward because the operation is actually acting on attached tenants, not tenants in secondary mode (even though the purpose of the functionality is to enable secondary mode to work).